legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/api/ApplicationQuery.java

```diff
@@ -180,6 +180,26 @@ public Response updateQuery(@PathParam("queryId") String queryId, Query query, @
         }
     }
 
+    @PUT
+    @Path("{queryId}/updateSelectedFields")
+    @ApiOperation(value = "Update query versionId")
+    @Consumes({MediaType.APPLICATION_JSON})
+    public Response updateQuerySelectedFields(@PathParam("queryId") String queryId, Query query, @ApiParam(hidden = true) @Pac4JProfileManager ProfileManager<CommonProfile> profileManager)
```

Comment:
Is the point of this endpoint to update only the fields of the query with `queryId` that are part of the `query` parameter? If so, I think it would make sense to make this endpoint a `PATCH` endpoint instead of a `PUT` endpoint, since we're only trying to update the fields present in the payload `query` object and not replace the entire query (https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH, https://www.geeksforgeeks.org/difference-between-put-and-patch-request/). Then, we could maybe rename this endpoint and rename the `queryStoreManager` function from `updateQuerySelectedFields` to `patchQuery`.
```suggestion
    @PATCH
    @Path("{queryId}")
    @ApiOperation(value = "Patch query")
    @Consumes({MediaType.APPLICATION_JSON})
    public Response patchQuery(@PathParam("queryId") String queryId, Query query, @ApiParam(hidden = true) @Pac4JProfileManager ProfileManager<CommonProfile> profileManager)
```

Reply:
<img width="383" alt="Screenshot 2024-03-04 at 11 30 38 AM" src="https://github.com/finos/legend-engine/assets/73408381/ca68a97a-a5da-41ae-b348-4893a56f28b9">
Thank you! However, it seems like Patch is not supported in engine's current dependency.

Reply:
Thank you! With current Engine's dependency. PATCH operation is not supported. Since we wrote our own update logic, do you think if it's okay with keeping use PUT? 
I've updated the function name to be patch instead

Reply:
Yeah if our dependency doesn't support `PATCH` operation then `PUT` is probably the next best option, but we can make the function called `patch` still since that's what it's actually doing.
Hopefully in the future if we update our dependencies and `PATCH` is supported, then we could update the method type.